src/Tokenizer.ts

```diff
@@ -0,0 +1,192 @@
+            throw new MethodParamsTokenizerException(`Invalid value ${value}`);
+
+        this.tokens.push(new MethodParamsToken(TokenType.PARAMETER_VALUE, value));
+        return valueEnd;
+    }
+
+    stringLiteral(idx:number):number {
+        const delimiter = this.raw[idx];
+        const matchStep = 2;
+        const matchedEnd = this._until(new RegExp(`([^\\\\]${delimiter})|\\n`), idx, matchStep);
+        if (this.raw.slice(matchedEnd, matchedEnd + matchStep).match(/\n/)) {
+            throw new MethodParamsTokenizerException(`Invalid multiline string literal.`);
+        }
+
+        const literalEnd = matchedEnd + matchStep;
+        if (this.raw[literalEnd - 1] !== delimiter) {
+            throw new MethodParamsTokenizerException(`Mismatched string literal delimiter ${delimiter}.`);
+        }
+        const literal = this.raw.slice(idx, literalEnd);
+        this.tokens.push(new MethodParamsToken(TokenType.PARAMETER_VALUE, literal));
+        return literalEnd;
+    }
+
+    _checkPrimitiveValue(value:string):boolean {
+        // Don't accept characters other than letters, digits, _, . and -
+        // It must have at least one letter, digit or _
+        const hasSomeChar = value.length > 0;
+        const invalidChar = !!value.match(/[^A-Z0-9_\.-]/i);
+        if (!hasSomeChar || invalidChar)
+            return false;
+        if (value.match(/[A-Z_]/i))
+            return this._checkNameValue(value);
+        return this._checkNumberValue(value);
+    }
+
+    _checkNameValue(value:string):boolean {
+        // A name must not start with digits, or have non word characters
+        return !value.match(/^\d/) && !value.match(/\W/);
+    }
+
+    _checkNumberValue(value:string):boolean {
+        // There should be a - at most once and always in the beginning of value
+        const minusPos = value.lastIndexOf('-');
+        if (minusPos > 0)
+            return false;
+        const positiveNumber =  minusPos === -1 ? value : value.slice(1);
+
+        if (value.match(/\./))
+            return this._checkPositiveFloatValue(positiveNumber);
+        return this._checkPositiveIntValue(positiveNumber);
+    }
+
+    _checkPositiveFloatValue(value:string):boolean {
+        // There should be a unique . separator with at least one algarism before and after
+        const dots = value.match(/\./g);
+        const dotIdx = value.indexOf('.');
+        return dots !== null && dots.length === 1 && dotIdx !== value.length - 1
+        && this._checkPositiveIntValue(value.slice(0, dotIdx))
+        && this._checkDecimalValue(value.slice(dotIdx + 1));
+    }
+
+    _checkPositiveIntValue(value:string):boolean {
+        return value.length > 0 && !value.match(/\D/) && (value.length === 1 || !value.match(/^0/));
+    }
+
+    _checkDecimalValue(value:string):boolean {
+        return value.length > 0 && !value.match(/\D/);
+    }
+
+    _ignore(ignore:RegExp, start:number, step = 1):number {
```

Comment:
I didn't find any call to his function with step != 1.

Reply:
Wouldn't it be preferable to unify the signature  of `_ignore` and `_until`?

Reply:
ok.